src/Stratis.Bitcoin.Features.Consensus/Rules/CommonRules/StratisHeaderVersionRule.cs

```diff
@@ -9,17 +13,35 @@ namespace Stratis.Bitcoin.Features.Consensus.Rules.CommonRules
     /// <summary>
     /// Checks if <see cref="StratisMain"/> network block's header has a valid block version.
     /// </summary>
-    [HeaderValidationRule(CanSkipValidation = true)]
-    public class StratisHeaderVersionRule : ConsensusRule
+    [HeaderValidationRule]
+    public class StratisHeaderVersionRule : HeaderVersionRule
     {
         /// <inheritdoc />
-        /// <exception cref="ConsensusErrors.BadVersion">Thrown if block's version is outdated.</exception>
+        /// <exception cref="ConsensusErrors.BadVersion">Thrown if block's version is outdated or otherwise invalid.</exception>
         public override Task RunAsync(RuleContext context)
         {
             Guard.NotNull(context.ConsensusTip, nameof(context.ConsensusTip));
 
             ChainedHeader chainedHeader = context.ValidationContext.ChainedHeader;
 
+            // A version of precisely 7 is what is currently generated by the legacy C++ nodes.
+
+            // The stratisX block validation rules mandate if (!IsProtocolV3(nTime)) && (nVersion > 7), then reject block.
+            // Further, if (IsProtocolV2(nHeight) && nVersion < 7), then reject block.
+            // And lastly, if (!IsProtocolV2(nHeight) && nVersion > 6), then reject block.
+
+            // Protocol version determination is based on either the block height or timestamp as shown:
+            // IsProtocolV2(nHeight) { return TestNet() || nHeight > 0; }
+            // IsProtocolV3(nTime) { return TestNet() || nTime > 1470467000; }
+
+            // The mainnet genesis block has nTime = 1470713393, so V3 is applied immediately and this supersedes V2.
+            // The block versions have therefore been version 7 since genesis on Stratis mainnet.
+
+            // Whereas BIP9 mandates that the top bits of version be 001. So a standard node should never generate
+            // block versions above 7 and below 0x20000000.
```

Comment:
I would also delete this paragraph because Stratis does not implement BIP nine

Reply:
It has no BIP9 deployments in the network definition, which I don't see as being the same thing as not implementing it.

Also, blocks on Stratis testnet are being generated with BIP9 version numbers.

Reply:
I see that this is because test network runs on C# node as well as the mainnet runs on stratisX nodes. Okay so this paragraph can probably stay as is. However the word mandates is probably little misleading. Never mind

So current Stratis network really does not implement BIP nine, but test network does because it runs on C# node